src/partition/repart.c

```diff
@@ -5178,6 +5220,32 @@ static int context_minimize(Context *context) {
         return 0;
 }
 
+static int parse_filter_partitions(const char *p) {
+        int r;
+
+        for (;;) {
+                _cleanup_free_ char *name = NULL;
+                GptPartitionType type;
+
+                r = extract_first_word(&p, &name, ",", EXTRACT_CUNESCAPE|EXTRACT_DONT_COALESCE_SEPARATORS);
+                if (r == 0)
+                        break;
+                if (r < 0)
+                        return log_error_errno(r, "Failed to extract partition designator: %s", optarg);
+
+                r = gpt_partition_type_from_string(name, &type);
```

Comment:
is the list of valid strings published in the manpage?
Would it be worth also supporting listing GUIDs from the DPS?

Reply:
`gpt_partition_type_from_string()` does both identifiers and raw GUIDs. I've added a link to the `Type=` docs which describes the possible values.